kclvm/driver/src/tests.rs

```diff
@@ -382,28 +383,42 @@ fn test_get_pkg_list() {
     );
 }
 
-#[cfg(test)]
-// Define a mock structure to simulate the behavior of Command::output
-struct MockCommand {
-    output: Result<MockCommandOutput, std::io::Error>,
+#[test]
+fn test_lookup_the_nearest_file_dir_for_update() {
+    let path = PathBuf::from(".")
+        .join("src")
+        .join("test_data")
+        .join("kpm_update");
+    let result = lookup_the_nearest_file_dir(path.clone(), "kcl.mod");
+    assert!(result.is_some());
+    assert_eq!(
+        result.unwrap().display().to_string(),
+        path.canonicalize().unwrap().display().to_string()
+    );
+
+    let main_path = path.join("subdir").join("main.k");
+    let result = lookup_the_nearest_file_dir(main_path, "kcl.mod");
+    assert!(result.is_some());
+    assert_eq!(
+        result.unwrap().display().to_string(),
+        path.canonicalize().unwrap().display().to_string()
+    );
 }
-// Define a mock structure to represent the output of Command::output
-struct MockCommandOutput {
-    status: std::process::ExitStatus,
-    stderr: Vec<u8>,
+
+fn test_fetch_mod_metadata() {
+    let path = PathBuf::from(".")
+        .join("src")
+        .join("test_data")
+        .join("kpm_update");
+
+    let update_mod = update_kcl_module(path.clone());
+    // Show more information when the test fails.
+    println!("{:?}", update_mod);
+    assert!(!update_mod.is_err());
 }
 
 #[test]
-fn test_update_kcl_module_failure() {
-    let manifest_path = PathBuf::from("path/to/manifest");
-    fn mock_command_new_failing(_command: &str) -> MockCommand {
-        MockCommand {
-            output: Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                "Command failed",
-            )),
-        }
-    }
-    let result = update_kcl_module(manifest_path);
-    assert!(result.is_err());
+fn test_update_module() {
```

Comment:
Hi @octonawish-akcodes  CI failed and would you please sign the DCO?

Reply:
im inspecting that, and i am having trouble signing dco due to conflicts :/